src/components/ContactsApp.jsx

```diff
@@ -73,9 +75,22 @@ SelectionBarWithActions.propTypes = {
 class ContactsApp extends React.Component {
   state = {
     displayedContact: null,
+    importation: null,
     isCreationFormDisplayed: false
   };
 
+  displayImportation = () => {
+    this.setState({
+      importation: Importation.init()
+    });
+  };
+
+  updateImportation = importation => {
+    this.setState({
```

Comment:
This will re-render the entire app, if t is what you need, it is ok, otherwise this shall not be under this component responsibility.

Reply:
Ok, I could move this to the `ContactImportationModal` and only deal with the modal visibility at the toplevel. Would it sound good?

Reply:
In a components tree, if some data or some behavior are not shared with another part of the tree, it is more readable to put these data or this behavior as close to the components that need them as possible. And it saves a lot of useless re-rendering (instead of implementing `shouldComponentUpdate` in every level of the components tree).

As soon as some data or some behavior are shared, you have two solutions:
- host them on the most recent ancestor ;
- use the flux architecture to expose those to the correct component.